libs/SalesforceSDK/src/com/salesforce/androidsdk/util/AuthConfigTask.java

```diff
@@ -64,8 +67,10 @@ public AuthConfigTask(AuthConfigCallbackInterface callback) {
 
     @Override
     protected Void doInBackground(Void... nothings) {
-        final String loginServer = SalesforceSDKManager.getInstance().getLoginServerManager().getSelectedLoginServer().url;
-        if (loginServer.equals(LoginServerManager.PRODUCTION_LOGIN_URL) || loginServer.equals(LoginServerManager.SANDBOX_LOGIN_URL)) {
+        final String loginServer = SalesforceSDKManager.getInstance().getLoginServerManager().getSelectedLoginServer().url.trim();
+        if (loginServer.equals(LoginServerManager.PRODUCTION_LOGIN_URL) ||
+                loginServer.equals(LoginServerManager.SANDBOX_LOGIN_URL) ||
+                !URLUtil.isHttpsUrl(loginServer) || HttpUrl.parse(loginServer) == null) {
```

Comment:
Validating that the URL is valid before attempting to hit the `.well-known` endpoint.